packages/core/src/api/blockManipulation/selections/selection.ts

```diff
@@ -216,3 +219,47 @@ export function setSelection(
   //  restriction that the start/end blocks must have content.
   tr.setSelection(TextSelection.create(tr.doc, startPos, endPos));
 }
+
+export function getSelectionCutBlocks(tr: Transaction) {
```

Comment:
This is likely over-complicated. There is probably a way to do this via shared parent depths

Reply:
Got it, I think an easier solution might be similar to what @matthewlipski built for `getSelection` above